components/scroll-wrapper/scroll-wrapper.js

```diff
@@ -0,0 +1,222 @@
+				left: -10px;
+				right: auto;
+			}
+
+			.d2l-scroll-wrapper-actions {
+				position: -webkit-sticky;
+				position: sticky;
+				top: 0;
+				z-index: 4;
+			}
+
+			.d2l-scroll-wrapper-button {
+				background-color: var(--d2l-color-regolith);
+				border: 1px solid var(--d2l-color-mica);
+				border-radius: 50%;
+				box-shadow: 0 2px 12px 0 rgba(0, 0, 0, 0.15);
+				box-sizing: content-box;
+				cursor: pointer;
+				display: inline-block;
+				font-family: inherit;
+				font-size: inherit;
+				height: 18px;
+				line-height: 0;
+				margin: 0;
+				padding: 10px;
+				position: absolute;
+				text-decoration: none;
+				top: 4px;
+				width: 18px;
+			}
+			.d2l-scroll-wrapper-button::-moz-focus-inner {
+				border: 0;
+			}
+			.d2l-scroll-wrapper-button:hover,
+			.d2l-scroll-wrapper-button:focus {
+				background-color: var(--d2l-color-sylvite);
+				outline-style: none;
+			}
+			.d2l-scroll-wrapper-button.focus-visible {
+				box-shadow: 0 0 0 2px #ffffff, 0 0 0 4px var(--d2l-color-celestine), 0 2px 12px 0 rgba(0, 0, 0, 0.15);
+			}
+			:host([scrollbar-right]) .d2l-scroll-wrapper-button-right {
+				display: none;
+			}
+			:host([scrollbar-left]) .d2l-scroll-wrapper-button-left {
+				display: none;
+			}
+
+			/* hide wrapper visuals from print view */
+			@media print {
+				.d2l-scroll-wrapper-actions {
+					display: none;
+				}
+				:host([h-scrollbar][show-actions]) .d2l-scroll-wrapper-container {
+					border-left: none;
+					border-right: none;
+				}
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.showActions = false;
+		this._container = null;
+		this._hScrollbar = true;
+		this._scrollbarLeft = false;
+		this._scrollbarRight = false;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this._container = this.shadowRoot.querySelector('.d2l-scroll-wrapper-container');
+		new ResizeObserver(
+			() => requestAnimationFrame(
+				() => this.checkScrollbar()
+			)
+		).observe(this._container);
+		requestAnimationFrame(() => this.checkScrollbar());
+	}
+
+	render() {
+		const actions = this.showActions ? html`
+			<div class="d2l-scroll-wrapper-actions">
+				<button class="d2l-scroll-wrapper-button d2l-scroll-wrapper-button-left" @click="${this._scrollLeft}" aria-hidden="true">
+					<d2l-icon icon="tier1:chevron-left"></d2l-icon>
+				</button>
+				<button class="d2l-scroll-wrapper-button d2l-scroll-wrapper-button-right" @click="${this._scrollRight}" aria-hidden="true">
+					<d2l-icon icon="tier1:chevron-right"></d2l-icon>
+				</button>
+			</div>` : null;
+		return html`
+			${actions}
+			<div class="d2l-scroll-wrapper-container" @scroll="${this._checkScrollThresholds}"><slot></slot></div>
+		`;
+	}
+
+	checkScrollbar() {
+		if (!this._container) return;
+		const hScrollbar = Math.abs(this._container.offsetWidth - this._container.scrollWidth);
+		this._hScrollbar = (hScrollbar > 0);
+		this._checkScrollThresholds();
+	}
+
+	scrollDistance(distance, smooth) {
```

Comment:
Is the ability to pass in the `smooth` parameter just to make the animation faster for tests?  Will the browser automatically know to "ignore" `smooth` in `scrollBy` if reduced motion is on?

Reply:
That's my understanding, yes. Safari doesn't support `smooth` at all, and obviously legacy-Edge isn't using it. I tested this out in Chrome and Firefox and neither seem to do anything differently when `prefers-reduced-motion` is chosen. 

I'm not super worried about this, since it's not really an animation so much as it's just normal scrolling, which also doesn't seem to behave differently with that setting. 🤷